### PR TITLE
fixes for race conditions when running tests from the root

### LINKED ISSUE #13578: createApp.test.ts is failing
## Expected Behavior

All tests passed.

## Actual Behavior

```
The test suite failed to run

ENOENT: no such file or directory, lstat '<YOUR_DIR>/backstage/node_modules/jest-worker/build/index.js'
```

## Steps to Reproduce

1. `cd packages/create-app`
2. `yarn test`
3. Press `a` to run all tests

## Context

I'm writing some additional test cases related to the issue https://github.com/backstage/backstage/issues/13408

When I tried to run all existing tests, it threw the error. After some research, I found [this question on SO](https://stackoverflow.com/questions/61430230/mock-fs-failed-with-jest-no-such-file-or-directory-lstat-and-mock-fs-1-de) that says this is related to `mockFs`.

I was able to fix the error by doing the following change:

```ts
// Before
afterAll(() => {
  mockFs.restore();
});

// After, all tests passed
afterEach(() => {
  mockFs.restore();
});
```

By the way, I'm using `Node v16.16.0` but I also tested on `v12` and `v14`.

I opened this issue to check if that's happening for everyone or just me, it can be something specific on my configs. Also, it can be happening to other tests.

## Your Environment

- Browser Information: Google Chrome Version 104.0.5112.101 (Official Build) (x86_64)

- Output of `yarn backstage-cli info`: 

```text
OS:   Darwin 21.6.0 - darwin/x64
node: v16.16.0
yarn: 3.2.3
cli:  0.19.0-next.2 (local)
backstage:  N/A

Dependencies:
  @backstage/app-defaults                                   0.0.0-use.local
  @backstage/backend-app-api                                0.0.0-use.local
  @backstage/backend-common                                 0.0.0-use.local
  @backstage/backend-defaults                               0.0.0-use.local
  @backstage/backend-plugin-api                             0.0.0-use.local
  @backstage/backend-tasks                                  0.0.0-use.local
  @backstage/backend-test-utils                             0.0.0-use.local
  @backstage/catalog-client                                 0.0.0-use.local, 1.0.4
  @backstage/catalog-model                                  0.0.0-use.local
  @backstage/cli-common                                     0.0.0-use.local
  @backstage/cli                                            0.0.0-use.local
  @backstage/codemods                                       0.0.0-use.local
  @backstage/config-loader                                  0.0.0-use.local
  @backstage/config                                         0.0.0-use.local
  @backstage/core-app-api                                   0.0.0-use.local
  @backstage/core-components                                0.0.0-use.local, 0.11.0
  @backstage/core-plugin-api                                0.0.0-use.local, 1.0.5
  @backstage/create-app                                     0.0.0-use.local
  @backstage/dev-utils                                      0.0.0-use.local
  @backstage/errors                                         0.0.0-use.local
  @backstage/integration-react                              0.0.0-use.local, 1.1.3
  @backstage/integration                                    0.0.0-use.local, 1.3.0
  @backstage/plugin-adr-backend                             0.0.0-use.local
  @backstage/plugin-adr-common                              0.0.0-use.local
  @backstage/plugin-adr                                     0.0.0-use.local
  @backstage/plugin-airbrake-backend                        0.0.0-use.local
  @backstage/plugin-airbrake                                0.0.0-use.local
  @backstage/plugin-allure                                  0.0.0-use.local
  @backstage/plugin-analytics-module-ga                     0.0.0-use.local
  @backstage/plugin-apache-airflow                          0.0.0-use.local
  @backstage/plugin-api-docs-module-protoc-gen-doc          0.0.0-use.local
  @backstage/plugin-api-docs                                0.0.0-use.local
  @backstage/plugin-apollo-explorer                         0.0.0-use.local
  @backstage/plugin-app-backend                             0.0.0-use.local
  @backstage/plugin-auth-backend                            0.0.0-use.local
  @backstage/plugin-auth-node                               0.0.0-use.local
  @backstage/plugin-azure-devops-backend                    0.0.0-use.local
  @backstage/plugin-azure-devops-common                     0.0.0-use.local
  @backstage/plugin-azure-devops                            0.0.0-use.local
  @backstage/plugin-badges-backend                          0.0.0-use.local
  @backstage/plugin-badges                                  0.0.0-use.local
  @backstage/plugin-bazaar-backend                          0.0.0-use.local
  @backstage/plugin-bazaar                                  0.0.0-use.local
  @backstage/plugin-bitbucket-cloud-common                  0.0.0-use.local
  @backstage/plugin-bitrise                                 0.0.0-use.local
  @backstage/plugin-catalog-backend-module-aws              0.0.0-use.local
  @backstage/plugin-catalog-backend-module-azure            0.0.0-use.local
  @backstage/plugin-catalog-backend-module-bitbucket-cloud  0.0.0-use.local
  @backstage/plugin-catalog-backend-module-bitbucket-server 0.0.0-use.local
  @backstage/plugin-catalog-backend-module-bitbucket        0.0.0-use.local
  @backstage/plugin-catalog-backend-module-gerrit           0.0.0-use.local
  @backstage/plugin-catalog-backend-module-github           0.0.0-use.local
  @backstage/plugin-catalog-backend-module-gitlab           0.0.0-use.local
  @backstage/plugin-catalog-backend-module-ldap             0.0.0-use.local
  @backstage/plugin-catalog-backend-module-msgraph          0.0.0-use.local
  @backstage/plugin-catalog-backend-module-openapi          0.0.0-use.local
  @backstage/plugin-catalog-backend                         0.0.0-use.local
  @backstage/plugin-catalog-common                          0.0.0-use.local, 1.0.5
  @backstage/plugin-catalog-graph                           0.0.0-use.local
  @backstage/plugin-catalog-graphql                         0.0.0-use.local
  @backstage/plugin-catalog-import                          0.0.0-use.local
  @backstage/plugin-catalog-node                            0.0.0-use.local
  @backstage/plugin-catalog-react                           0.0.0-use.local, 1.1.3
  @backstage/plugin-catalog                                 0.0.0-use.local
  @backstage/plugin-cicd-statistics-module-gitlab           0.0.0-use.local
  @backstage/plugin-cicd-statistics                         0.0.0-use.local
  @backstage/plugin-circleci                                0.0.0-use.local
  @backstage/plugin-cloudbuild                              0.0.0-use.local
  @backstage/plugin-code-climate                            0.0.0-use.local
  @backstage/plugin-code-coverage-backend                   0.0.0-use.local
  @backstage/plugin-code-coverage                           0.0.0-use.local
  @backstage/plugin-codescene                               0.0.0-use.local
  @backstage/plugin-config-schema                           0.0.0-use.local
  @backstage/plugin-cost-insights-common                    0.0.0-use.local
  @backstage/plugin-cost-insights                           0.0.0-use.local
  @backstage/plugin-dynatrace                               0.0.0-use.local
  @backstage/plugin-explore-react                           0.0.0-use.local
  @backstage/plugin-explore                                 0.0.0-use.local
  @backstage/plugin-firehydrant                             0.0.0-use.local
  @backstage/plugin-fossa                                   0.0.0-use.local
  @backstage/plugin-gcalendar                               0.0.0-use.local
  @backstage/plugin-gcp-projects                            0.0.0-use.local
  @backstage/plugin-git-release-manager                     0.0.0-use.local
  @backstage/plugin-github-actions                          0.0.0-use.local
  @backstage/plugin-github-deployments                      0.0.0-use.local
  @backstage/plugin-github-issues                           0.0.0-use.local
  @backstage/plugin-github-pull-requests-board              0.0.0-use.local
  @backstage/plugin-gitops-profiles                         0.0.0-use.local
  @backstage/plugin-gocd                                    0.0.0-use.local
  @backstage/plugin-graphiql                                0.0.0-use.local
  @backstage/plugin-graphql-backend                         0.0.0-use.local
  @backstage/plugin-home                                    0.0.0-use.local, 0.4.24
  @backstage/plugin-ilert                                   0.0.0-use.local
  @backstage/plugin-jenkins-backend                         0.0.0-use.local
  @backstage/plugin-jenkins-common                          0.0.0-use.local
  @backstage/plugin-jenkins                                 0.0.0-use.local
  @backstage/plugin-kafka-backend                           0.0.0-use.local
  @backstage/plugin-kafka                                   0.0.0-use.local
  @backstage/plugin-kubernetes-backend                      0.0.0-use.local
  @backstage/plugin-kubernetes-common                       0.0.0-use.local
  @backstage/plugin-kubernetes                              0.0.0-use.local
  @backstage/plugin-lighthouse                              0.0.0-use.local
  @backstage/plugin-newrelic-dashboard                      0.0.0-use.local
  @backstage/plugin-newrelic                                0.0.0-use.local
  @backstage/plugin-org                                     0.0.0-use.local
  @backstage/plugin-pagerduty                               0.0.0-use.local
  @backstage/plugin-periskop-backend                        0.0.0-use.local
  @backstage/plugin-periskop                                0.0.0-use.local
  @backstage/plugin-permission-backend                      0.0.0-use.local
  @backstage/plugin-permission-common                       0.0.0-use.local, 0.6.3
  @backstage/plugin-permission-node                         0.0.0-use.local
  @backstage/plugin-permission-react                        0.0.0-use.local, 0.4.4
  @backstage/plugin-proxy-backend                           0.0.0-use.local
  @backstage/plugin-rollbar-backend                         0.0.0-use.local
  @backstage/plugin-rollbar                                 0.0.0-use.local
  @backstage/plugin-scaffolder-backend-module-cookiecutter  0.0.0-use.local
  @backstage/plugin-scaffolder-backend-module-rails         0.0.0-use.local
  @backstage/plugin-scaffolder-backend-module-yeoman        0.0.0-use.local
  @backstage/plugin-scaffolder-backend                      0.0.0-use.local
  @backstage/plugin-scaffolder-common                       0.0.0-use.local
  @backstage/plugin-scaffolder                              0.0.0-use.local
  @backstage/plugin-search-backend-module-elasticsearch     0.0.0-use.local
  @backstage/plugin-search-backend-module-pg                0.0.0-use.local
  @backstage/plugin-search-backend-node                     0.0.0-use.local
  @backstage/plugin-search-backend                          0.0.0-use.local
  @backstage/plugin-search-common                           0.0.0-use.local, 1.0.0
  @backstage/plugin-search-react                            0.0.0-use.local
  @backstage/plugin-search                                  0.0.0-use.local
  @backstage/plugin-sentry                                  0.0.0-use.local
  @backstage/plugin-shortcuts                               0.0.0-use.local
  @backstage/plugin-sonarqube-backend                       0.0.0-use.local
  @backstage/plugin-sonarqube                               0.0.0-use.local
  @backstage/plugin-splunk-on-call                          0.0.0-use.local
  @backstage/plugin-stack-overflow-backend                  0.0.0-use.local
  @backstage/plugin-stack-overflow                          0.0.0-use.local, 0.1.4
  @backstage/plugin-tech-insights-backend-module-jsonfc     0.0.0-use.local
  @backstage/plugin-tech-insights-backend                   0.0.0-use.local
  @backstage/plugin-tech-insights-common                    0.0.0-use.local
  @backstage/plugin-tech-insights-node                      0.0.0-use.local
  @backstage/plugin-tech-insights                           0.0.0-use.local
  @backstage/plugin-tech-radar                              0.0.0-use.local
  @backstage/plugin-techdocs-addons-test-utils              0.0.0-use.local
  @backstage/plugin-techdocs-backend                        0.0.0-use.local
  @backstage/plugin-techdocs-module-addons-contrib          0.0.0-use.local
  @backstage/plugin-techdocs-node                           0.0.0-use.local
  @backstage/plugin-techdocs-react                          0.0.0-use.local
  @backstage/plugin-techdocs                                0.0.0-use.local
  @backstage/plugin-todo-backend                            0.0.0-use.local
  @backstage/plugin-todo                                    0.0.0-use.local
  @backstage/plugin-user-settings                           0.0.0-use.local
  @backstage/plugin-vault-backend                           0.0.0-use.local
  @backstage/plugin-vault                                   0.0.0-use.local
  @backstage/plugin-xcmetrics                               0.0.0-use.local
  @backstage/release-manifests                              0.0.0-use.local
  @backstage/test-utils                                     0.0.0-use.local
  @backstage/theme                                          0.0.0-use.local
  @backstage/types                                          0.0.0-use.local
  @backstage/version-bridge                                 0.0.0-use.local
```

### PR DESCRIPTION
**Pull request by Rugvip**
## Hey, I just made a Pull Request!

Some fixes extracted from #13574

Fixes #13578

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by benjdlambert — APPROVED** [review]

File: packages/create-app/src/lib/tasks.test.ts [hunk 1]
```diff
@@ -30,6 +29,55 @@ import {
...
+    '@backstage/plugin-search-react': '1.0.0',
+    '@backstage/plugin-tech-radar': '1.0.0',
+    '@backstage/plugin-techdocs': '1.0.0',
+    '@backstage/plugin-techdocs-react': '1.0.0',
+    '@backstage/plugin-techdocs-module-addons-contrib': '1.0.0',
+    '@backstage/plugin-user-settings': '1.0.0',
+    '@backstage/theme': '1.0.0',
+    '@backstage/test-utils': '1.0.0',
+  },
+}));
```
Comment (freben) [inline packages/create-app/src/lib/tasks.test.ts:79]: Long list is long

Comment (freben) [inline packages/create-app/src/lib/tasks.test.ts:79]: Long list is long

File: packages/backend-common/src/database/DatabaseManager.ts [hunk 2]
```diff
@@ -352,7 +352,9 @@ export class DatabaseManager {
     let lastKeepaliveFailed = false;
 
     setInterval(() => {
-      client.raw('select 1').then(
+      // During testing it can happen that the environment is torn down and
+      // this client is `undefined`, but this interval is still run.
+      client?.raw('select 1').then(
```
Comment (freben) [inline packages/backend-common/src/database/DatabaseManager.ts:357]: What. How can it change its value?

**Review by freben — APPROVED** [review]

File: packages/backend-common/src/database/DatabaseManager.ts (hunk 2 above)
Comment (Rugvip) [reply packages/backend-common/src/database/DatabaseManager.ts:357]: image

image

I have no idea 😅

Comment (freben) [reply packages/backend-common/src/database/DatabaseManager.ts:357]: Computers were a mistake.

**Pull request merged**